### PR TITLE
Fixing dupe background-size rule for bg icons;

### PR DESCRIPTION
**Pull request by seangates**
## Description
- removes the `background-size: contain;` rule from `.background-icon-base()`
- adds a new Less mixin guard using a reversed logical check for `.background-icon-image()` setting `background-size: contain;`

## Context
_**Note:** Found during work on #614, and needed to be addressed before some modules can be refactored._

When using `.background-icon-base()` there was an assumption that `.background-icon-image()` was always going to be in a different selector. But, since both `.background-icon-base()` and `.background-icon-image()` both had a `background-size` rule, they could potentially be used in the _same_ selector, which would cause a lint issue complaining of a duplicate rule.

To fix this, we create a new Less mixin guard that is opposite of the existing guard, and we're able to add only one `background-size` rule per `.background-icon-image()` mixin.

Because this touches a lot of files, please see me if you have questions. Each dist file should now have the `background-size` removed (or moved from certain selectors), depending on how the guard was evaluated. Therefore, I may be able to direct you on what to look for, and clear up confusion.

### CODE REVIEW

**Comment by ianmcburnie** [comment]
Not looked in full yet, but would this be a breaking change for users of .background-icon-base()?

**Comment by seangates** [comment]
@ianmcburnie So, I've gone through, and found one instance where this _might_ break (Notice). I'll get it sorted and let you know.

**Comment by ianmcburnie** [comment]
I don't mean for internal stuff (but that needs checking too), I mean if they are using the mixin in their own app.

**Comment by ianmcburnie** [comment]
If it is a breaking change, which I think it might be because removing `background-size: contain;` could have unintended consequences for an app (what do you think?), we either do a major version (would prefer to avoid that) or create a new alias which we switch over to internally and then deprecate the old one.

**Comment by ianmcburnie** [comment]
Also starting to think we should remove these type of mixins from our public API. I did remove the vast majority of "utility" mixins and vars, but some still remain.

**Comment by seangates** [comment]
Yeah, unless we have good reason, we probably don't need to have these be exposed.

Let me create a new alias so we can deprecate the old one. 😞

File: src/less/less/mixins-common.less [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
 
+// DEPRECATED - do not use
+.background-icon-base() {
+    .background-icon-image-base();
+
+    background-size: contain;
+}
+
 // base styles for BEM svg.icon block
 .foreground-icon-base() {
```
Comment (ianmcburnie) [inline src/less/less/mixins-common.less]: Should we also create a new alias for foreground icon to match? i.e.`foreground-icon-image-base()` Or not worth the hassle (considering we will be hiding these away in Skin v8)?

File: src/less/less/mixins-common.less [hunk 2]
```diff
@@ -1,13 +1,19 @@
 @import "variables-common.less";
 
 // base styles for BEM span.icon block
-.background-icon-base() {
+.background-icon-image-base() {
```
Comment (ianmcburnie) [inline src/less/less/mixins-common.less]: Should we hide this away from our public API now? (just leaving the old one until v8?)

File: src/less/less/mixins-common.less (hunk 1 above)
Comment (seangates) [reply src/less/less/mixins-common.less]: Thoughts on the best way to "hide" it?

Comment (ianmcburnie) [reply src/less/less/mixins-common.less]: Yeah, let's wait.

File: src/less/less/mixins-common.less (hunk 2 above)
Comment (seangates) [reply src/less/less/mixins-common.less]: Thoughts on the best way to "hide" it?

Comment (seangates) [reply src/less/less/mixins-common.less]: Thoughts on the best way to "hide" it?

Comment (ianmcburnie) [reply src/less/less/mixins-common.less]: Hmm, let's wait also. We also have plenty of time before 7.3.0 goes out I just realised. So we can do more refactoring as necessary.

**Pull request merged**